proposals/0292-package-registry-service.md

```diff
@@ -384,160 +744,171 @@ and individual commits may be cryptographically signed by authors.
 Unless you can establish a direct connection between
 an artifact and a commit in a source tree,
 there's no way to determine the provenance of that artifact.
-However,
-a [transparent log] of checksums or the use of digital signatures
-can provide similar non-repudiation guarantees.
 
-### Information disclosure
+Source archives generated by [`git-archive(1)`]
+include the checksum of the `HEAD` commit as a comment.
+If the history of a project is available
+and the commit used to generate the source archive is signed with [GPG],
+the cryptographic signature may be used to verify the authenticity.
 
-An attacker could scrape public code repositories
-for `Package.swift` files that use hardcoded credentials in dependency URLs,
-and attempt to reuse those credentials to impersonate the user.
+```terminal
+$ git rev-parse HEAD
+b7c37c81f164e5dce0f64e3d75c79a48fb1fe00b3
 
-```swift
-dependencies: [
-  .package(name: "TopSecret",
-           url: "https://<token>:x-oauth-basic@github.com/mona/TopSecret",
-           checksum: "2c4a4ce92225fb766447c1757abb916e13f68eba0459f1287ee62e4941d89bbf")
-]
+$ swift package archive-source -o LinkedList-1.2.0.zip
+Generated LinkedList-1.2.0.zip
+
+$ zipnote LinkedList-1.2.0.zip | grep "@ (zip file comment below this line)" -A 1 | tail -n 1
+b7c37c81f164e5dce0f64e3d75c79a48fb1fe00b3
+
+$ git verify-commit b7c37c81f164e5dce0f64e3d75c79a48fb1fe00b3
+gpg: Signature made Tue Dec 16 00:00:00 2020 PST
+gpg:                using RSA key BFAA7114B920808AA4365C203C5C1CF
+gpg: Good signature from "Mona Lisa Octocat <mona@noreply.github.com>" [ultimate]
 ```
 
-This kind of attack can be mitigated on an individual basis
-by using an unauthenticated URL and setting a mirror.
+Otherwise,
+a checksum database and the use of digital signatures
+can both provide similar non-repudiation guarantees.
+
+### Information disclosure
+
+A user may inadvertently reveal the existence of a private registry
+or expose hardcoded credentials
+by checking in their project's `.swiftpm/config` file.
+
+An attacker could scrape public code repositories for `.swiftpm/config` files
+and attempt to reuse those credentials to impersonate the user.
+
+```json
+{
+  "registries": {
+      "default": {
+        "url": "https://<USERNAME>:<TOKEN>@swift.pkg.github.com/<OWNER>/"
+      }
+  },
+  "version": 1
+}
 
-```terminal
-$ swift package config set-mirror \
-    --original-url https://github.com/mona/TopSecret \
-    --mirror-url https://<token>:x-oauth-basic@github.com/mona/TopSecret
 ```
 
+This kind of attack can be mitigated on an individual basis
+by adding `.swiftpm/config` to a project's `.gitignore` file.
```

Comment:
probably all of `./.swiftpm/config`?

Reply:
probable this should also be added to the `.gitignore` template  in `swift package init`

Reply:
I agree. That's what I suggest in the following sentence.

> The risk could be mitigated for all users
> if Swift Package Manager included a `.gitignore` file
> in its new project template.